src/include/mscclpp/mscclpp_nccl.h

```diff
@@ -0,0 +1,64 @@
+/*************************************************************************
+ * Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * See LICENSE.txt and NOTICES.txt for license information
+ ************************************************************************/
+
+#ifndef MSCCLPP_NCCL_H_
+#define MSCCLPP_NCCL_H_
+
+#include "nccl.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct mscclpp_ncclComm* mscclpp_ncclComm_t;
+
+typedef struct { char internal[NCCL_UNIQUE_ID_BYTES]; } mscclpp_ncclUniqueId;
+
+/* Generates an Id to be used in mscclpp_ncclCommInitRank. mscclpp_ncclGetUniqueId should be
+ * called once and the Id should be distributed to all ranks in the
+ * communicator before calling mscclpp_ncclCommInitRank. */
+extern ncclResult_t  (*mscclpp_ncclGetUniqueId)(mscclpp_ncclUniqueId* uniqueId);
+
+/* Creates a new communicator (multi thread/process version).
+ * rank must be between 0 and nranks-1 and unique within a communicator clique.
+ * Each rank is associated to a CUDA device, which has to be set before calling
+ * mscclpp_ncclCommInitRank.
+ * mscclpp_ncclCommInitRank implicitly syncronizes with other ranks, so it must be
+ * called by different threads/processes or use mscclpp_ncclGroupStart/mscclpp_ncclGroupEnd. */
```

Comment:
This references a "CUDA" device, as well as mscclpp_ncclGroupStart and mscclpp_ncclGroupEnd, which we don't use.